analyses/snv-callers/util/wrangle_functions.R

```diff
@@ -0,0 +1,260 @@
+  # Args:
+  #   maf_df: maf data.frame that has been turned into a data.frame and has
+  #           WXS mutations filtered using `wxs_bed_filter` function.
+  #   wgs_size: genome size in bp to be used for WGS samples
+  #   wxs_size: genome size in bp to be used for WGS samples
+  #
+  # Make a genome size variable
+  tmb <- maf_df %>%
+    dplyr::mutate(genome_size = dplyr::recode(experimental_strategy,
+      "WGS" = wgs_size,
+      "WXS" = wxs_size
+    )) %>%
+    dplyr::group_by(
+      Tumor_Sample_Barcode, experimental_strategy, genome_size,
+      short_histology
+    ) %>%
+    # Count number of mutations for that sample
+    dplyr::summarize(mutation_count = dplyr::n()) %>%
+    # Calculate TMB
+    dplyr::mutate(tmb = mutation_count / (genome_size / 1000000))
+
+  return(tmb)
+}
+
+annotr_maf <- function(maf_df, annotation_obj = annotations, bp_window = 0) {
+  # Annotate the genomic regions mutations are in from a MAF formatted data.
+  # frame with AnnotatR object.
+  #
+  # Args:
+  #   maf_df: maf data that has been turned into a data.frame.
+  #   annotation_obj: an annotation object with the desired annotations.
+  #   bp_window: how many base pairs away can it be from the BED region to still
+  #              be included? Default is 0 bp. This argument gets forwarded
+  #              to GenomicRanges::findOverlaps's `maxgap` argument.
+  #
+  # Read in the genomic regions annotation object
+  annotations <- readr::read_rds(annot_rds)
+
+  # Use custom function to turn our MAF data into a GRanges
+  maf_ranges <- maf_to_granges(maf_df)
+
+  # Intersect the regions we read in with the annotations
+  annot_matches <- GenomicRanges::findOverlaps(annotation_obj, maf_ranges,
+    maxgap = bp_window
+  )
+
+  # Extract the annotation from the regions that overlap the mut
+  annot <- data.frame(
+    annotation_obj[annot_matches@from]@elementMetadata@listData,
+    maf_ranges[annot_matches@to]@elementMetadata@listData,
+    stringsAsFactors = FALSE
+  ) %>%
+    dplyr::rename_at(dplyr::vars(dplyr::starts_with("mcols.")), substr, 7, 10000) %>%
+    dplyr::mutate("type" = as.factor(gsub("^hg38_genes_", "", type)))
+
+  return(annot)
+}
+
+find_cosmic_overlap <- function(maf_df, bp_window = 0) {
+  # Find how much overlap a MAF formatted data.frame has with COSMIC mutations
+  # Return a column in the MAF data.frame that says whether or not it is also
+  # in the COSMIC mutations set.
+  #
+  # Args:
+  #   maf_df: MAF formatted data that has been turned into a data.frame and has
+  #           been run through `set_up_variables`
+  #   bp_window: how many base pairs away can it be from the BED region to still
+  #              be included? Default is 0 bp. This argument gets forwarded
+  #              to GenomicRanges::findOverlaps's `maxgap` argument.
+  #
+  # Turn both these datasets into GRanges
+  maf_granges <- maf_to_granges(maf_df)
+  cosmic_granges <- maf_to_granges(suppressMessages(readr::read_tsv(cosmic_clean_file)))
```

Comment:
Is there no argument to `readr::read_tsv` that will accomplish the same thing as `suppressMessages`?

Reply:
Where is `cosmic_clean_file` coming from? Is your plan to have it be in your global environment when you use this function? This can be sourced _anywhere_ so `cosmic_clean_file` should be passed as an argument to this function.

Reply:
> Is there no argument to `readr::read_tsv` that will accomplish the same thing as `suppressMessages`?

I had looked in the docs to see if there was an argument for this and didn't see anything but Dr. Google suggests that `message = FALSE` should do the trick.

Reply:
Dr. Google lied. This does not work.

Reply:
Dr. Google! :shakes fist:

Reply:
This as an argument works: `col_type = readr::cols()`.